### PR TITLE
Kamu 117 implement auxillary view component for block hashes display

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/components/display-hash/display-hash.component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+@Component({
+    selector: "app-display-hash",
+    templateUrl: "./display-hash.component.html",
+    styleUrls: ["./display-hash.component.sass"],
+})
+export class DisplayHashComponent {
+    @Input() public value: string;
+    @Input() public datasetInfo: DatasetInfo;
+    @Input() public showCopyButton?: boolean = false;
```
Comment (zaychenko-sergei) [inline src/app/components/display-hash/display-hash.component.ts:14]: ? not necessary, default value is enough

File: src/app/components/display-hash/display-hash.component.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Input() public class?: string;
+    @Input() public isNavigable?: boolean = true;
```
Comment (zaychenko-sergei) [inline src/app/components/display-hash/display-hash.component.ts:16]: ? not necessary, default value is enough

File: src/app/components/display-hash/display-hash.component.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@Component({
+    selector: "app-display-hash",
+    templateUrl: "./display-hash.component.html",
+    styleUrls: ["./display-hash.component.sass"],
+})
+export class DisplayHashComponent {
+    @Input() public value: string;
+    @Input() public datasetInfo: DatasetInfo;
+    @Input() public showCopyButton?: boolean = false;
+    @Input() public class?: string;
```
Comment (zaychenko-sergei) [inline src/app/components/display-hash/display-hash.component.ts:15]: Maybe a default class instead?

File: src/app/components/display-hash/display-hash.component.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        this.clipboard.copy(text);
+        if (event.currentTarget !== null) {
+            const currentElement: HTMLButtonElement =
+                event.currentTarget as HTMLButtonElement;
+            const currentElementChildren: HTMLCollectionOf<HTMLElement> =
+                currentElement.children as HTMLCollectionOf<HTMLElement>;
+            setTimeout(() => {
+                currentElementChildren[0].style.display = "inline-block";
+                currentElementChildren[1].style.display = "none";
+                currentElement.classList.remove("clipboard-btn--success");
```
Comment (zaychenko-sergei) [inline src/app/components/display-hash/display-hash.component.ts:45]: Declare constants for repeating class names

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]

File: src/app/components/display-hash/display-hash.component.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/components/display-hash/display-hash.component.ts:16]: Fixed

File: src/app/components/display-hash/display-hash.component.ts (hunk 4 above)
Comment (dmitriy-borzenko) [reply src/app/components/display-hash/display-hash.component.ts:45]: Declared constants for repeating class names

File: src/app/components/display-hash/display-hash.component.ts (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/components/display-hash/display-hash.component.ts:15]: Added default class

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**